config/passport.js

```diff
@@ -1,28 +1,65 @@
-import passport from "passport";
-import strategies from "passport-local";
-import models from '../models/';
+import passport from 'passport';
+import strategies from 'passport-local';
+import { Strategy as JWTStrategy, ExtractJwt } from 'passport-jwt';
+import dotenv from 'dotenv';
+import models from '../models';
+
+dotenv.config();
 
 const LocalStrategy = strategies.Strategy;
-const User = models.User;
+const { User } = models;
 
-passport.use(
+passport.use('signup',
   new LocalStrategy(
     {
-      usernameField: "user[email]",
-      passwordField: "user[password]"
+      usernameField: User.email,
+      passwordField: User.password
     },
-    function(email, password, done) {
-      User.findOne({ email: email })
-        .then(function(user) {
-          if (!user || !user.validPassword(password)) {
-            return done(null, false, {
-              errors: { "email or password": "is invalid" }
-            });
-          }
+    async (email, password, done) => {
+      try {
+        const user = await User.findOne({ where: { email: email } });
+        if(!email || !password) {
+          return done(null, false, { message: 'The email' });
+        }
+        if(user) {
+          return done(null, false, { message: 'The email is already taken' });
+        }
+        return done(null, email);
+      }catch (err) {
+        return done(err);
+      }
+    }
+  )
+);
 
-          return done(null, user);
-        })
-        .catch(done);
+/* 
+* The passport strategy to authorize and authenticate the user using JWT token
+*  Them middleware will check first the token in headers
+* Check if the user with id in the payloads exists in db.
+*/
+passport.use(
+  'jwt',
+  new JWTStrategy(
+    {
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken('Authorization'),
+      secretOrKey: process.env.SECRET,
```

Comment:
Did you mean secretKey here or "secret0rkey"

Reply:
Thank you @aaronsekisambu, the secretOrKey is a string or buffer containing the secret or PEM-encoded public key so the key work is secretOrKey.